### PR TITLE
Authorize /lgtm and /approve with OWNERS file

### LINKED ISSUE #44: Support OWNERS files
**Is your feature request related to a problem? Please describe.**
I want to specify exactly who can provide reviews and approve pull requests since there are a lot of people in the organization who do not qualify as reviewers and approvers.

**Describe the solution you'd like**
I'd be happy with any way to specify a named set of people or teams that can be reviewers and approvers. Either by reading the [OWNERS](https://www.kubernetes.dev/docs/guide/owners/) file (like prow does), or something more simple like just passing the action a list of names in the workflow file. I realize the full logic of owners files is pretty complex, and I just care about having a bit more control than just org membership.

**Describe alternatives you've considered**
N/A

**Additional context**
I'm happy to submit a PR for this if this is a welcome feature.

### PR DESCRIPTION
**Pull request by carolynvs**
When an OWNERS file is present, the /lgtm and /approve commands are authorized using the contents of the file instead of org/repo membership. The github action does not checkout the repository itself to locate OWNERS. The workflow is responsible for checking out the repo first before calling the prow action.

The format of the OWNERS file is a subset of prow's OWNERS file. I am only looking for the file at the root of the repository. The format is as follows:

```yaml
approvers:
- user1
- user2

reviewers:
- user1
- user2
- user3
```

/lgtm checks if the commenter is in the reviewers section, and /approve checks if the commenter is in the approvers section. Membership in approvers does not authorize the use of /lgtm.

Closes #44

### CODE REVIEW

**Comment by carolynvs** [comment]
If you'd like to see where this was tested, I've been chatting with the prow action on this test PR: https://github.com/ladygogo/test/pull/1

**Comment by jpmcb** [comment]
Oooo excellent! Thanks for this

Planning to get eyes on this week. Again, thank you so so much for your contribution!

**Comment by carolynvs** [comment]
I have been a bit stumped by the one test failure. I'm unsure what I changed that is causing it.

```
FAIL __tests__/main.test.ts
  ● runs with no options

    Command failed: node /home/runner/work/prow-github-actions/prow-github-actions/lib/main.js

      10 |   }
      11 | 
    > 12 |   expect(cp.execSync(`node ${ip}`, options).toString()).toContain(
         |             ^
      13 |     'not yet supported'
      14 |   )
      15 | })

 ```

**Review by jpmcb — CHANGES_REQUESTED** [review]
This is looking amazing!! 👏🏼 

Thank you so so much for this huge feature add.

I'm seeing a few `prettier` errors being thrown on running `npm run all`. I think once those are addressed, we can get this in

**Comment by jpmcb** [comment]
> I have been a bit stumped by the one test failure. I'm unsure what I changed that is causing it.

The github actions are actually broken for some reason. That's been a long standing problem that I haven't had a ton of time to look into. For example, it's [broken in dependabot bumps](https://github.com/jpmcb/prow-github-actions/pull/50). I'll make a new issue.

Running the tests locally look good for me:

```
npm run test-all

...

Test Suites: 25 passed, 25 total
Tests:       6 skipped, 81 passed, 87 total
Snapshots:   0 total
Time:        7.411 s
Ran all test suites.
```

**Comment by carolynvs** [comment]
Alright, prettier has been appeased! Thanks for pointing that out. I'm new to js/ts and am still figuring out what all I should be running locally.

File: src/utils/auth.ts
Comment (jpmcb) [inline src/utils/auth.ts:6]: This is great and works well, but I'm wondering if we could think about using octokit's `getContents` API: https://octokit.github.io/rest.js/v18#repos-get-content

We do have this as an existing paradigm in the pr labeler: https://github.com/jpmcb/prow-github-actions/blob/main/src/cronJobs/prLabeler.ts#L153-L169

When a user providers a `.github/labels.yaml` or `.github/labels.yml` we dynamically get it via the API to use in the PR labeler. This is nice so we don't have to clone the users repo into their actions runner to get this to work.

File: src/utils/auth.ts
Comment (jpmcb) [inline src/utils/auth.ts:232]: nit: Can we name this `getOwnersFilePath`?

File: src/utils/auth.ts
Comment (jpmcb) [inline src/utils/auth.ts:245]: nit: Can we name this `isInOwnersFile`?

File: src/utils/auth.ts
Comment (carolynvs) [reply src/utils/auth.ts:6]: Hah, sorry I talked myself out of what I just suggested. 😀  I'll switch to retrieving the OWNERS file directly through the API, and if we decide to do multi-owner file support per repo later, we can switch it up.

Not having to clone the repo first is a heck of a lot easier.

**Comment by carolynvs** [comment]
I've updated the PR to use "OwnersFile" instead of just "Owners" all by itself, and to retrieve the owners file via the API instead of cloning the repo.

**Comment by carolynvs** [comment]
@jpmcb When you have time to review, let me know if there are more changes I need to make. I have a patch ready that replies to the PR when you don't have permission to run /lgtm or /approve so that people know when things aren't working.

**Review by jpmcb — APPROVED** [review]
This is great. Let's bring this in!

**Pull request merged**
